src/core/text/TextMetrics.js

```diff
@@ -89,6 +89,50 @@ export default class TextMetrics
         );
     }
 
+    /**
+     * Check whether a byte is an emoji character or not.
+     *
+     * @param {number} charCode - the byte to test.
+     * @param {number} nextCharCode - the possible second byte of the emoji.
+     * @return {number} 0 means not a emoji, 1 means single byte, 2 means double bytes.
+     */
+    static isEmojiChar(charCode, nextCharCode)
```

Comment:
Overall, lgtm.
1. I'd wonder, as an API that could be used externally, whether "charCode" and "nextCharCode" could be "firstCharCode" and "secondCharCode". I know internally, where it's used, it's the current one and the next one, but I could see it being useful to fire 2 characters at this static function in my games.
2. For other pixi peeps - I'd want https://github.com/pixijs/pixi.js/pull/4545 merged in before this please